lib/reporting/reports/sales_tax/sales_tax_totals_by_order.rb

```diff
@@ -0,0 +1,196 @@
+# frozen_string_literal: true
+
+# rubocop:disable Metrics/ClassLength
+module Reporting
+  module Reports
+    module SalesTax
+      class SalesTaxTotalsByOrder < Base
+        def search
+          report_line_items.orders
+        end
+
+        def order_permissions
+          return @order_permissions unless @order_permissions.nil?
+
+          @order_permissions = ::Permissions::Order.new(user, ransack_params)
+        end
+
+        def report_line_items
+          # Needed to filter by supplier_id
+          @report_line_items ||= Reporting::LineItems.new(order_permissions, params)
+        end
+
+        def query_result
+          # We'll group the line items by
+          # [tax_rate_id, order_id]
+          orders = report_line_items.list.map(&:order).uniq
+          orders.flat_map(&join_tax_rate)
+            .group_by(&group_key)
+            .map(&change_root_to_order)
+        end
```

Comment:
Nice way to break up the complexity into separate procs (I've not seen it done that way before).
There's still some complex massaging of data, which makes me wonder if we can simplify it. But I can't think of anything better at the moment. 

Perhaps it would be worth having a unit test to test this more closely (eg under `/spec/lib/reports/`). But I think it's better to thoroughly test the report output, which you've already done 👍